src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -6,13 +6,35 @@
 import * as vscode from 'vscode'
 import { CLOUDWATCH_LOGS_SCHEME } from '../shared/constants'
 import { fromExtensionManifest } from '../shared/settings'
+import { Any } from '../shared/utilities/typeConstructors'
 import { CloudWatchLogsGroupInfo } from './registry/logStreamRegistry'
 import { CloudWatchLogsParameters } from './registry/logStreamRegistry'
 
 // URIs are the only vehicle for delivering information to a TextDocumentContentProvider.
 // The following functions are used to structure and destructure relevant information to/from a URI.
 // Colons are not valid characters in either the group name or stream name and will be used as separators.
 
+/**
+ * This function concatenates the path and query to create a unique (enough) identifier for a URI.
+ * @param uri
+ * @returns
+ */
+export function uriToKey(uri: vscode.Uri): string {
+    // We sort the query object by key because queries can be identical semantically in different order.
+    // `Any` is used as type hack to avoid errors when recreating the object.
+    if (uri.query) {
+        const queryObject = JSON.parse(uri.query)
+        const orderedObject = Object.keys(queryObject)
+            .sort()
+            .reduce(function (acc: Any, key: string) {
+                acc[key] = queryObject[key]
+                return acc
+            }, {})
+        return uri.path + JSON.stringify(orderedObject)
+    }
```

Comment:
oh, there's a lot going on here :)

1. I forgot that `query` is just a JSON blob we're passing around.
    - Does the JSON object have an id that is meaningful to us, that could be used as part of the key? Then we don't need to deal with hashing the entire thing.
    - can JSON web tokens be used to hash a js object? https://stackoverflow.com/a/30855168
        - does nodejs support jwt? (don't want a new dependency for this.)
2. I forgot that `vscode.Uri.query` is just a string, not a dictionary.
3. If we are going to use a JSON string as the key, we might want to declare that as part of the caller requirements, i.e. `uri.query` already has the form we want here.
4. This is non-trivial so would need some tests. But that should wait until we decide on how to build a "key" from the query